libcob/fileio.c

```diff
@@ -4764,32 +4764,17 @@ cob_file_close (cob_file_api *a, cob_file *f, const int opt)
 				f->file_pid = 0;
 				return COB_STATUS_00_SUCCESS;
 			}
-#ifdef _MSC_VER /* explicit only stream or fd close with this compiler */
-			if (f->file != NULL) {
-				fclose ((FILE *)f->file);
-				f->file = NULL;
-#ifdef _WIN32
-				/* at least on MSVC that closes the underlying file descriptor, too */
-				f->fd = -1;
-#endif
-			}
-#endif
-		} else {
-			if (f->fd >= 0) {
-				close (f->fd);
-				f->fd = -1;
-#ifdef _MSC_VER
-				f->file = NULL;
-#endif
-			}
 		}
-#ifndef _MSC_VER /* explicit disallowed there, is that the same for other compilers? */
 		if (f->file != NULL) {
+			/* This should close the underlying fd */
```

Comment:
When we drop all these parts, then `if (f->organization == COB_ORG_LINE_SEQUENTIAL) {` should be dropped completely as the code left is for pipes only, and those are always lsq.